### PR TITLE
Fetch and save logged in practitioner details

### LINKED ISSUE #1282: [Quest/eCBIS] Sync to server after creation of removal of family or family member fails.
**Describe the bug**
- When you create a family member or a new family and then try to sync you will notice a _Sync failed notification_ 
- The logs show that this is failing due to a malformed Bundle but I can't seem to find the Bundle in the local SQLite App DB. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to any family profile and add a family member
2. Click on the sync button and wait till sync is complete
3. Click sync on another device to check on the family member you added. 

**Expected behaviour**
- [ ]  Any time a change is made on any of the resources and sync is clicked, then all the local resources should be pushed to the server and the remote resources pushed to the device. 

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: Emulator
 - Android version: latest eCBIS release from the **_implement-generic-app-register_** branch

**Additional context**
AppID : `ecbis-saa`
Username: `ecbis`

### PR DESCRIPTION
**Pull request by ellykits**
Signed-off-by: Elly Kitoto <junkmailstoelly@gmail.com>

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1282

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by pld — DISMISSED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginScreen.kt [hunk 1]
```diff
@@ -251,20 +251,31 @@ fun LoginPage(
...
-                .testTag(LOGIN_ERROR_TEXT_TAG)
-          )
-        }
-        Spacer(modifier = modifier.height(40.dp))
+        Text(
+          fontSize = 14.sp,
+          color = MaterialTheme.colors.error,
+          text =
+            when (loginErrorState) {
+              LoginErrorState.NO_ERROR -> ""
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginScreen.kt:259]: do we need the no_error state? could we handle this with an `else` for the `when`?

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginScreen.kt:259]: then whenever we set to this constant we'd set to null instead

Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginScreen.kt:259]: We do not need it actually. I will make the `loginErrorState` parameter nullable.

Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginScreen.kt]: I have explicitly handled `null` condition in the `when` statement; we may in future need to further categorize the error messages.

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginScreen.kt]: ok good call, we wouldn't want to bury an error if it doesn't explicitly match the when conditions

**Pull request merged**